src/Functions/UserDefined/UserDefinedSQLFunctionVisitor.cpp

```diff
@@ -19,24 +22,96 @@ namespace ErrorCodes
     extern const int UNSUPPORTED_METHOD;
 }
 
-void UserDefinedSQLFunctionMatcher::visit(ASTPtr & ast, Data &)
+void UserDefinedSQLFunctionVisitor::visit(ASTPtr & ast)
 {
-    auto * function = ast->as<ASTFunction>();
-    if (!function)
+    const auto visit_child_with_shared_ptr = [&](ASTPtr & child)
```

Comment:
Would be good to have a comment why we are working with shared_ptr.

Reply:
could you help me with this one
I'm not sure how it affects the logic here, in some cases AST nodes will have an extra shared_ptr to their children.